src/Features/Core/Portable/Diagnostics/CodeAnalysisDiagnosticAnalyzerService.cs

```diff
@@ -58,11 +69,25 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
                 case WorkspaceChangeKind.SolutionCleared:
                 case WorkspaceChangeKind.SolutionReloaded:
                 case WorkspaceChangeKind.SolutionRemoved:
+
                     _analyzedProjectIds.Clear();
+                    _clearedProjectIds.Clear();
+
+                    // Let LSP know so that it requests up to date info, and will see our cached info disappear.
+                    _diagnosticsRefresher.RequestWorkspaceRefresh();
                     break;
             }
         }
 
+        public void Clear()
+        {
+            // Clear the list of analyzed projects.
+            _clearedProjectIds.AddRange(_analyzedProjectIds);
```

Comment:
The issue with this approach is you might have Run Code Analysis on the entire solution, but running a Build command on a single project clears out RCA diagnostics for entire solution, which seems undesirable. I think it's probably better to take the approach mentioned in https://github.com/dotnet/roslyn/pull/70412/files#r1363027754 and handle this for individual projects